sonar-plugin-api/src/main/java/org/sonar/api/batch/debt/DebtRemediationFunction.java

```diff
@@ -85,25 +80,32 @@ public boolean equals(Object o) {
     if (o == null || getClass() != o.getClass()) {
       return false;
     }
+
     DebtRemediationFunction that = (DebtRemediationFunction) o;
-    return new EqualsBuilder()
-      .append(type, that.type())
-      .append(coefficient, that.coefficient())
-      .append(offset, that.offset())
-      .isEquals();
+    if (type != that.type) {
+      return false;
+    }
+    if (coefficient != null ? !coefficient.equals(that.coefficient) : that.coefficient != null) {
+      return false;
+    }
+    return offset != null ? offset.equals(that.offset) : that.offset == null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS864)